lib/src/command/outdated.dart

```diff
@@ -55,9 +61,8 @@ class OutdatedCommand extends PubCommand {
     argParser.addOption('mark',
         help: 'Highlight packages with some property in the report.',
         valueHelp: 'OPTION',
-        allowed: ['outdated', 'none'],
-        defaultsTo: 'outdated',
-        hide: true);
+        allowed: ['null-safety', 'outdated', 'none'],
```

Comment:
It looks like the `mark` option already exists but is being un-hidden?

It may not be immediately clear to the user what the `mark` option is for. Perhaps rename to something like `mode`? It would then read: `--mode=null-safety`.

What does the `none` value do? Would user's ever send this in?

If we only ever really expect mode values of `null-safety` and `outdated`, perhaps just add a non-negatable `--null-safety` flag on the outdated command.

Reply:
> It looks like the mark option already exists but is being un-hidden?

Yes - it was a planned feature but with very limited use until now. (At most you could disable the colorization of versions). It was hidden in https://github.com/dart-lang/pub/pull/2403 after the suggestion at https://github.com/dart-lang/pub/issues/2395 .

> It may not be immediately clear to the user what the mark option is for. Perhaps rename to something like mode? It would then read: --mode=null-safety.

`mark` is not perfect, but the mental picture is that it goes through all the versions found by outdated, and marks those that fulfill some criterium. It is not a mode of operation for the outdated command in general. (There are many modes: `--prereleases`, `--dependency-overrides`, `--up-to-date` are all what I would call modes of outdated).

> What does the none value do? Would user's ever send this in?

`none` disables highlighting. I'm not sure I have very concrete use-cases, but one could think that someone finds the colors distracting, or want to produce a 'clean' report for including somewhere.

It doesn't cost us much to provide this option.
 
If we only ever really expect mode values of

> If we only ever really expect mode values of null-safety and outdated, perhaps just add a non-negatable --null-safety flag on the outdated command (and drop the mark option altogether).

We have some other ideas for markers that could be added in the future. Possible candidates:

* Marking versions on pub.dev with a specific tag.
* Marking versions using a specific language version (not only 2.10 for null-safety).
* Marking versions that are not broken

Reply:
After discussing this a bit with @jonasfj and @mit-mit I think I see what I got wrong here.

I thought we only had to change the presentation of the versions. But to be useful the option also has to show a different set of packages (All that are not currently null-safety enabled).

After realizing this `--mode` makes a lot more sense.

Thanks.